hail/python/hail/experimental/vcf_combiner.py

```diff
@@ -3,14 +3,58 @@
 import hail as hl
 from hail import MatrixTable, Table
 from hail.expr import StructExpression
-from hail.expr.expressions import expr_call, expr_array, expr_int32
+from hail.expr.expressions import expr_bool, expr_call, expr_array, expr_int32, expr_str
 from hail.genetics.reference_genome import reference_genome_type
 from hail.ir import Apply, TableMapRows, TopLevelReference
 from hail.typecheck import oneof, sequenceof, typecheck
 
 _transform_rows_function_map = {}
 _merge_function_map = {}
 
+@typecheck(string=expr_str, has_non_ref=expr_bool)
+def parse_as_ints(string, has_non_ref):
+    ints = string.split(r'\|')
+    ints = hl.cond(has_non_ref, ints[:-1], ints)
+    return ints.map(lambda i: hl.cond((hl.len(i) == 0) | (i == '.'), hl.null(hl.tint32), hl.int32(i)))
+
+@typecheck(string=expr_str, has_non_ref=expr_bool)
+def parse_as_doubles(string, has_non_ref):
+    ints = string.split(r'\|')
+    ints = hl.cond(has_non_ref, ints[:-1], ints)
+    return ints.map(lambda i: hl.cond((hl.len(i) == 0) | (i == '.'), hl.null(hl.tfloat64), hl.float64(i)))
+
+@typecheck(string=expr_str, has_non_ref=expr_bool)
+def parse_as_sb_table(string, has_non_ref):
+    ints = string.split(r'\|')
+    ints = hl.cond(has_non_ref, ints[:-1], ints)
+    return ints.map(lambda xs: xs.split(",").map(hl.int32))
+
+@typecheck(string=expr_str, has_non_ref=expr_bool)
+def parse_as_ranksum(string, has_non_ref):
+    typ = hl.ttuple(hl.tfloat64, hl.tint32)
+    items = string.split(r'\|')
+    items = hl.cond(has_non_ref, items[:-1], items)
+    return items.map(lambda s: hl.cond(
+        (hl.len(s) == 0) | (s == '.'),
+        hl.null(typ),
+        hl.rbind(s.split(','), lambda ss: hl.cond(
+            hl.len(ss) != 2,  # bad field, possibly 'NaN', just set it null
+            hl.null(hl.ttuple(hl.tfloat64, hl.tint32)),
+            hl.tuple([hl.float64(ss[0]), hl.int32(ss[1])])))))
+
+_as_function_map = {
+    'AS_QUALapprox': parse_as_ints,
+    'AS_RAW_MQ': parse_as_doubles,
+    'AS_RAW_MQRankSum': parse_as_ranksum,  # FIXME |NaN| issue
+    'AS_RAW_ReadPosRankSum': parse_as_ranksum,  # FIXME |NaN| issue
```

Comment:
Did you intend to fix these now? Maybe make a Github issue about this and assign to yourself if not.

Reply:
Oh, no, that's been fixed. Will amend.